common/scripted_effects/babylon_spontaneous_civil_war.txt

```diff
@@ -2,36 +2,224 @@
+	every_scope_state = {
+		limit = {
+			is_capital = no
+			is_incorporated = yes
+			NOT = {
+				has_variable = dynamic_country_cede
+			}
+			any_neighbouring_state = {
+				is_incorporated = yes
+				has_variable = dynamic_country_cede
+			}
+		}
+		random = {
+			chance = $STREMGTH$
+			set_variable = dynamic_country_cede
 		}
 	}
-	random_country = {
+	every_scope_state = {
 		limit = {
-			has_variable = civil_war_country
+			is_capital = no
+			is_incorporated = yes
+			NOT = {
+				has_variable = dynamic_country_cede
+			}
+			any_neighbouring_state = {
+				is_incorporated = yes
+				has_variable = dynamic_country_cede
+			}
+		}
+		random = {
+			chance = $STREMGTH$
+			set_variable = dynamic_country_cede
 		}
-		save_scope_as = civil_war_country
 	}
 	every_scope_state = {
 		limit = {
 			is_capital = no
+			is_incorporated = yes
+			NOT = {
+				has_variable = dynamic_country_cede
+			}
+			any_neighbouring_state = {
+				is_incorporated = yes
+				has_variable = dynamic_country_cede
+			}
 		}
 		random = {
-			chance = $CHANCE$
-			set_state_owner = scope:civil_war_country
+			chance = $STREMGTH$
+			set_variable = dynamic_country_cede
+		}
+	}
+	set_variable = is_ceding_dynamic_country_states
+	create_dynamic_country = {
+		origin = root
+		cede_state_trigger = {
+			owner = {
+				has_variable = is_ceding_dynamic_country_states
+			}
+			has_variable = dynamic_country_cede
+		}
+		color = grey
+		primary_unit_color = grey
+		secondary_unit_color = grey
+		tertiary_unit_color = grey
+		on_created = {
+			set_variable = civil_war_country
+		}
+	}
+	remove_variable = is_ceding_dynamic_country_states
+	random_country = {
+		limit = {
+			has_variable = civil_war_country
 		}
+		save_scope_as = civil_war_country
 	}
 	scope:civil_war_country = {
 		create_diplomatic_play = {
 			type = dp_revolution
 			target_country = root
 		}
 		remove_variable = civil_war_country
+		every_scope_state = {
+			remove_variable = dynamic_country_cede
+		}
+	}
+}
+
+# Scope: imploding country
+implode_country = {
+	set_variable = is_discord_implosion_country
+	every_scope_state = {
+		limit = {
+			is_capital = no
+		}
+		save_scope_as = dynamic_country_capital
+		# Try to get a pop that actually has a sizeable population in the state. Checks for 50%+ religion, 40%, 30% and then just takes whatever is there.
+		# Should always have a homeland in the selected state
+		if = {
+			limit = {
+				any_scope_pop = {
+					prev.state_region = {
+						is_homeland = prev.culture
+					}
+					prev = {
+						religion_percent_state = {
+							target = prev.religion
+							value >= 0.5
+						}
+					}
+				}
+			}
+			random_scope_pop = {
+				limit = {
+					prev.state_region = {
+						is_homeland = prev.culture
+					}
+					prev = {
+						religion_percent_state = {
+							target = prev.religion
+							value >= 0.5
+						}
+					}
+				}
+				religion = {
+					save_scope_as = new_religion_scope
+				}
+				culture = {
```

Comment:
it saves the culture of the pop with the biggest religion, so it definitely is possible to culture switch

Reply:
But only if that culture has a homeland there.